framework/src/org/checkerframework/common/value/ValueVisitor.java

```diff
@@ -51,29 +79,60 @@ public Void visitAnnotation(AnnotationTree node, Void p) {
         }
 
         Element element = TreeInfo.symbol((JCTree) node.getAnnotationType());
-        if (!(element.toString().equals(ArrayLen.class.getName())
+
+        if (element.toString().equals(IntRange.class.getName())) {
+            if (args.size() == 2
+                    && args.get(0).getKind() == Kind.ASSIGNMENT
+                    && args.get(1).getKind() == Kind.ASSIGNMENT) {
+                ExpressionTree expFrom;
+                ExpressionTree expTo;
+
+                if (((AssignmentTree) args.get(0)).getVariable().toString().equals("from")) {
+                    expFrom = ((AssignmentTree) args.get(0)).getExpression();
+                    expTo = ((AssignmentTree) args.get(1)).getExpression();
+                } else {
+                    expFrom = ((AssignmentTree) args.get(1)).getExpression();
+                    expTo = ((AssignmentTree) args.get(0)).getExpression();
+                }
+
+                if (isIntLiteral(expFrom.getKind()) && isIntLiteral(expTo.getKind())) {
```

Comment:
Here, too, if the arguments are not literals, that seems to be grounds for a different error, since the `@IntRange` annotation is ill-formed.